src/ReservoirRouter.sol

```diff
@@ -0,0 +1,291 @@
+pragma solidity 0.8.13;
+
+import { ERC20 } from "solmate/tokens/ERC20.sol";
+import { FixedPointMathLib } from "solmate/utils/FixedPointMathLib.sol";
+import { Math } from "@openzeppelin/utils/math/Math.sol";
+
+import { IReservoirRouter, ExtraData } from "src/interfaces/IReservoirRouter.sol";
+import { IReservoirPair } from "v3-core/src/interfaces/IReservoirPair.sol";
+import { StablePair } from "v3-core/src/curve/stable/StablePair.sol";
+
+import { ReservoirLibrary } from "src/libraries/ReservoirLibrary.sol";
+import { TransferHelper } from "src/libraries/TransferHelper.sol";
+
+import { PeripheryImmutableState } from "src/abstract/PeripheryImmutableState.sol";
+import { PeripheryPayments } from "src/abstract/PeripheryPayments.sol";
+import { PredicateHelper } from "src/abstract/PredicateHelper.sol";
+import { Multicall } from "src/abstract/Multicall.sol";
+
+contract ReservoirRouter is
+    IReservoirRouter,
+    PeripheryImmutableState,
+    PeripheryPayments,
+    PredicateHelper,
+    Multicall
+{
+    uint256 constant MINIMUM_LIQUIDITY = 1e3;
+
+    constructor (address aFactory, address aWETH) PeripheryImmutableState(aFactory, aWETH)
+    {}
+
+    function _addLiquidity(
+        address tokenA,
+        address tokenB,
+        uint curveId,
+        uint amountADesired,
+        uint amountBDesired,
+        uint amountAMin,
+        uint amountBMin
```

Comment:
Are we going to write the whole periphery library using Uniswap style?

- Braces & spacing
- Variable namings

I would propose adopting our style for variable names but leaving the braces & spacing, or we could just use `forge fmt` and never look back :P

Reply:
`forge fmt` doesn't do variable names.... 

Okay let's leave the braces and spacing, but adopt our variable names (`aArg`, `lVar`, `rRet`)

Reply:
fixed